graphbolt/src/csc_sampling_graph.cc

```diff
@@ -781,10 +782,16 @@ inline int64_t LaborPick(
     std::iota(picked_data_ptr, picked_data_ptr + num_neighbors, offset);
     return num_neighbors;
   }
-  torch::Tensor heap_tensor = torch::empty({fanout * 2}, torch::kInt32);
+#define STACK_SIZE_LABOR 1024
   // Assuming max_degree of a vertex is <= 4 billion.
-  auto heap_data = reinterpret_cast<std::pair<float, uint32_t>*>(
-      heap_tensor.data_ptr<int32_t>());
+  std::array<std::pair<float, uint32_t>, STACK_SIZE_LABOR> heap_stack;
```

Comment:
just stack? heap? it is a little bit confusing. Maybe a more descriptive name?

Reply:
Storage is from stack but the datastructure is heap. Maybe heap_from_stack?

Reply:
Just heap should be enough, and L788 makes more sense `heap_data = heap.data();`